File: test/fib.pl
#!/usr/bin/env perl
sub fib {
    my $n = shift;
    return $n if $n <= 1;
    return fib($n-1) + fib($n-2);
}

for my $i (0..10) {
    print "fib($i) = " . fib($i) . "\n";
}

File: test/fib.lua
function fib(n)
    if n <= 1 then
        return n
    end
    return fib(n - 1) + fib(n - 2)
end

for i = 0, 10 do
    print(string.format("fib(%d) = %d", i, fib(i)))
end
